src/lib.rs

```diff
@@ -12,10 +12,11 @@ use pyo3::{prelude::*, sync::GILOnceCell};
 #[macro_use]
 mod py_gc;
 
+pub mod errors;
```

Comment:
how come we have a bunch of `pub` stuff in private modules that isnt reexported anywhere?

Reply:
The code should end up being used by Python wrappers e.g. `#[pyclass]` / `#[pyfunction]` etc

(Do you mean everything should really be `pub(crate)`?)

Reply:
No I just forgot how weird the module system is in rust :laughing: 

```rust
mod foo {
    pub struct Bar;
}

mod baz {
    use crate::foo::Bar;
}
```
This compiles which I think is weird.... but that's okay I guess